### PR TITLE
feat: convert k8s submissions from pods to jobs (#9296)

### PR DESCRIPTION
**Pull request by stoksc**
This change updates our Kubernetes resource manager to submit one job per Determined task instead of many pods. This is a complicated change but we think it is worth it because:
- Jobs play nice with resource quotas and other Kubernetes features out of the box.
- Eventually we can delegate restarts, TTL, pause/resume (using suspend), and more to jobs.
- They allow us to better integrate with Kueue and other tools in the ml ecosystem.
- Supporting VolcanoJobs (or similar alternatives) alongside Jobs is realistic.
- The refactor is net positive w.r.t. test coverage (20% to 80%) and code quality.

This commit is the result of several PRs, enumerated here for easier discovery.
- #9296 
- #9443 
- #9447 
- #9450 
- #9451 

## Description
This is a feature branch. It already contains the core of the change, but we'll be merging release notes, docs, extra tests, and maybe some more adjustments (nothing major) before it lands later this week.

## Checklist

- [x] Changes have been manually QA'd
- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [x] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by stoksc** [comment]
As of [66f9d1e](https://github.com/determined-ai/determined/pull/9438/commits/66f9d1ec4a79f298dccc847eeb3f27febaae41fd) CI is very happy so this should be a good starting point for any last minute refactors.

**Review by carolinaecalderon — APPROVED** [review]
LGTM; I've already approved all the commits individually

File: master/internal/db/postgres_test_utils.go
Comment (stoksc) [inline master/internal/db/postgres_test_utils.go]: this is a revert of these changes. i just did `git checkout upstream/main -- master/internal/db/postgres_test_utils.go`

**Review by azhou-determined — APPROVED** [review]
python file lgtm

**Comment by stoksc** [comment]
once tests pass i'm going to land this without an explicit approval from @tara-det-ai since she approved the docs pr that landed into this.

**Review by tara-det-ai — APPROVED** [review]
LGTM

**Pull request merged**
